### ISSUE TITLE
Check fails when there is no issue 1

### ISSUE THREAD

**Issue by ebouchut** [Levi-Lesches/blocking-issues#29]
When a repository does not have an issue with the number 1, then the `blocking-issue` action/check fails because of this, according to the error message:

> Error: Got an HTTP 403 error while retrieving issue #1

See: https://github.com/ebouchut/Trio-dev-docs/actions/runs/11998558054/job/33445449252?pr=1#step:2:7

FYI: I

- started with issues disabled, 
- created a PR (with the number 1), 
- got this error when the check ran and failed
- I then manually created the `blocked` label
- enabled issues
- created a new issue (with the number 2)
- updated PR 1's description
- got the same error
- Seems like a 🐣 🥚 error :-)

.

**Comment by Levi-Lesches** [comment]
Interesting. Seems to me like it should have resolved itself after you enabled issues. 

Your repo is private -- can you make it public? Or if you want to keep it private, make a new repo and try to reproduce? 

If you do make it public, can you try to edit the description again? Seeing HTTP 403 and not 404 makes me think this may have more to do with permissions than it does with missing issue numbers

**Comment by ebouchut** [comment]
Context: I use a private repository and a GitHub free plan.

I cannot make this repository public.

Meanwhile, I looked into the permissions side as you suggested.
✅ 🥳 I figured out that **adding write permissions for `checks`, `issues`, and `pull-requests`** fixed this problem.

```yaml
# [...]

jobs:
  blocking_issues:
    permissions:
      checks: write
      issues: write
      pull-requests: write
    # [...]
```



Here is the updated workflow.

```yaml
name: Prevent Blocked PRs from being Merged

on:
  pull_request_target:
    types:
      - opened
      - edited
      - closed
      - reopened
  issues:
    types:
      - opened
      - edited
      - deleted
      - transferred
      - closed
      - reopened

jobs:
  blocking_issues:
    permissions:
      checks: write
      issues: write
      pull-requests: write
    runs-on: ubuntu-latest
    steps:
      - uses: Levi-Lesches/blocking-issues@v2
```

**Comment by ebouchut** [comment]
Quick question is `checks: write` superfluous?

**Comment by jpcenteno** [comment]
I had the same problem and fixed it doing the same as @ebouchut did!

1. I enabled Read and Write permissions from the settings page.
2. Granted write permissions for issues and pull-requests in the yaml file.

Note that I didn't grant `checks: write` permissions. I don't know what checks are BTW.

---

One thing that I noticed is that the job exits with an **Error** status, but everything works fine (The tag and the comment).

```
Getting current issue...
Processing #1[6](https://github.com/jpcenteno/personal-website-nextjs-implementation/actions/runs/12540130906/job/34967070636#step:2:7)
Initializing labels...
#16 is blocked by 14,15
Writing comment...
Updating label...
Is blocked? true
Error: Issue/PR is blocked.
```

I granted `checks: write` just to see if that was the issue and it threw an error message on the PR discussion:

```
Status: Error ⚠️
Issues blocking this PR:
#NaN ⚠️ Issue/PR not found
```

Undoing this change fixed the message.

----

Is returning an error the expected behavior when marking an issue as blocked? I'm not so worried about this thing since everything works. Just courious.

Sorry that I can't share the repo I'm working on. It's a private one.

**Comment by Levi-Lesches** [comment]
>  I don't know what checks are BTW.

Good question -- a check is an action or workflow that runs specifically in the context of "checking" a PR. In our case, this action is considered a check because it can block a PR from being merged. So, 

> Is returning an error the expected behavior when marking an issue as blocked?

Yes, because that's the signal to GitHub that the PR should not be merge-able. Note that the actual function called is [`core.setFailed()`](https://github.com/actions/toolkit/tree/main/packages/core#exit-codes), which means it's more of an intentional failure than an unexpected error. 

> Quick question is checks: write superfluous?

> I granted checks: write just to see if that was the issue and it threw an error message on the PR discussion:

Interesting that it helped in one case and caused an error in another. My guess is you _should_ include this line, because when an issue is closed, the bot goes back to all other failed checks and updates them, which would require write permissions on checks. @jpcenteno I'd love to see more of that `checks: write` error output if you still have it. 

----

Since you both got this working in the end, I'll consider this a documentation issue: I'll update the README to mention that for private repos, you should include

```yaml
 permissions:
      checks: write
      issues: write
      pull-requests: write
```

with a note to try removing `checks: write` if you still have issues.

**Comment by jpcenteno** [comment]
First of all, I want to thank both of you for developing this action. It adds a missing feature that I've always wanted.

---

On granting `checks: write` permissions:

> I granted checks: write just to see if that was the issue and it threw an error message on the PR discussion:

> I'd love to see more of that checks: write error output if you still have it.

Here is the output from the job that ran after I first enabled `checks: write`.

```
Run Levi-Lesches/blocking-issues@87d4122edf89dc9314d141649b0159a0e4c425d7
  with:
    use-label: Blocked Issue
    token: ***
Getting current issue...
Processing #[2](https://github.com/jpcenteno/xyz/actions/runs/12540011601/job/34966819031#step:2:2)0
Initializing labels...
#20 is blocked by NaN
Error: Issue not found: #NaN
Writing comment...
Found old comment (id 256488714[3](https://github.com/jpcenteno/xyz/actions/runs/12540011601/job/34966819031#step:2:3)). Updating...
Updating label...
Is blocked? false
Issue/PR is not blocked.
```

After giving it more thought, I realized that the problem might be that the PR wasn't associated with any issue, but I was unable to reproduce the issue.

I tested it with another PR that was linked to an issue, and your GitHub action worked as expected. Then, I cleared the PR's description to unlink it. The job successfully unblocked the PR and updated the message correctly.

**Comment by Levi-Lesches** [comment]
```console
#20 is blocked by NaN
Error: Issue not found: #NaN
```

Finally reproduced this! It seems you have, somewhere in your PR's description, the words `blocked by: #` but no number. For example, see [this issue](https://github.com/Levi-Lesches/blocking-issues-test/issues/41) and [this error](https://github.com/Levi-Lesches/blocking-issues-test/actions/runs/12554888518/job/35004223429): 

```md
Blocked by: #hello
```
```console
Run Levi-Lesches/blocking-issues@main
Getting current issue...
Processing #41
Initializing labels...
#41 is blocked by NaN
Error: Issue not found: #NaN
Writing comment...
Updating label...
Is blocked? false
Issue/PR is not blocked.
```

It makes sense that clearing the description got it to work. I've updated the code and README, publishing now

**Comment by Levi-Lesches** [comment]
Fixed the regex parsing and updated the README in [version 2.0.4](https://github.com/marketplace/actions/blocking-issues?version=v2.0.4) -- thanks again for reporting and debugging!

I've updated the `v2` tag to include the fix, but you can always specify a version manually: 

```yaml
    steps:
      - uses: Levi-Lesches/blocking-issues@v2.0.4
 ```
